code/game/machinery/ftl_shieldgen.dm

```diff
@@ -206,3 +225,59 @@
 	if(get_dir(src, target) == in_dir)
 		return 1
 	return 0
+
+/obj/effect/ftl_shield/Initialize(var/newloc)
+	..(newloc)
+	spawn(1) // Updates neightbors after we're gone.
+		for(var/direction in GLOB.cardinals)
+			var/turf/T = get_step(src.loc, direction)
+			if(T)
+				for(var/obj/effect/ftl_shield/F in T)
+					F.update_icon()
+
+/obj/effect/ftl_shield/attack_hand(var/mob/living/user)
+	impact_effect(3) // Harmless, but still produces the 'impact' effect.
+	..()
+
+/obj/effect/ftl_shield/Bumped(atom/A)
+	..(A)
+	impact_effect(2)
+
+
+/obj/effect/ftl_shield/update_icon(var/update_neightbors = 0)
+	overlays.Cut()
+	var/list/adjacent_shields_dir = list()
+	for(var/direction in GLOB.cardinals)
+		var/turf/T = get_step(loc, direction)
+		if(T) // Incase we somehow stepped off the map.
+			for(var/obj/effect/ftl_shield/F in T)
+				if(update_neightbors)
+					F.update_icon(0)
+				adjacent_shields_dir |= direction
+				break
+	if(density)
+		icon_state = "shield"
+		set_light(3, 3, "#66FFFF")
+	else
+		icon_state = "shield_broken"
+		set_light(3, 5, "#FF9900")
+
+	// Edge overlays
+	for(var/found_dir in adjacent_shields_dir)
+		add_overlay(image(src.icon, src, "shield_edge", dir = found_dir))
+
+
+// Small visual effect, makes the shield tiles 	brighten up by becoming more opaque for a moment, and spreads to nearby shields.
+obj/effect/ftl_shield/proc/impact_effect(var/i, var/list/affected_shields = list())
+	alpha = 175
+	animate(src, alpha = initial(alpha), time = 1)
+	affected_shields |= src
+	i--
+	if(i)
+		spawn(2)
+			for(var/direction in GLOB.cardinals)
+				var/turf/T = get_step(src, direction)
+				if(T) // Incase we somehow stepped off the map.
+					for(var/obj/effect/ftl_shield/F in T)
```

Comment:
thats not a list of shields that a list of directions with adjacent shields u tit